File: 08.JS-Applications/Exams/JS Applications Exam - Cars Tube/src/services/userService.js
import * as request from './request.js';
import * as authService from './authService.js';

export const login = async (username, password) => {
    const user = await request.post('/users/login', { username, password });
    authService.saveUser(user);
    return user;
}

export const register = async (username, password) => {
    const user = await request.post('/users/register', { username, password });
    authService.saveUser(user);
    return user;
}

export const logout = async () => {
    const res = await request.get('/users/logout');
    authService.removeUser();
    return res;
}

File: 11.C#-Web-Basics/Exams/C# Web Development Basics Exam - 20 Feb 2022/FootballManager/Services/UserService.cs
using FootballManager.Contracts;
using FootballManager.Data.Common;
using FootballManager.Data.Models;
using FootballManager.ViewModels.Users;
using System.Security.Cryptography;
using System.Text;

namespace FootballManager.Services
{
    public class UserService : IUserService
    {
        private readonly IRepository repo;

        private readonly IValidationService validationService;

        public UserService(
            IRepository repo,
            IValidationService validationService)
        {
            this.repo = repo;
            this.validationService = validationService;
        }

        public string Login(LoginViewModel model)
        {
            var user = this.repo.All<User>()
                .Where(u => u.Username == model.Username)
                .Where(u => u.Password == HashPassword(model.Password))
                .SingleOrDefault();

            return user?.Id;
        }

        public bool Register(RegisterViewModel model)
        {
            var registered = false;
            var isValid = validationService.ValidateModel(model);

            if (!isValid)
            {
                return isValid;
            }

            var user = new User()
            {
                Username = model.Username,
                Email = model.Email,
                Password = HashPassword(model.Password)
            };

            try
            {
                repo.Add<User>(user);
                repo.SaveChanges();
                registered = true;
            }
            catch (Exception)
            {
            }

            return registered;
        }

        private string HashPassword(string password)
        {
            byte[] passwordArray = Encoding.UTF8.GetBytes(password);

            using (SHA256 sha256 = SHA256.Create())
            {
                return Convert.ToBase64String(sha256.ComputeHash(passwordArray));
            }
        }
    }
}